src/components/MainPage/MainPage.js

```diff
@@ -0,0 +1,123 @@
+                })
+                .then(result => {
+                    setAllTasks(result.data);
+                })
+            } catch(e) {
+                setAlert({
+                    isOpen: true,
+                    text: e.message,
+                });
+                setIsAuth(false);
+                localStorage.clear();
+            };
+        }
+
+        getTasks();
+    }, []);
+
+    const logout = async () => {
+        try {
+            await axios.get('http://localhost:8000/logout', 
+            {
+                withCredentials: true, 
+                credentials: 'include'
+            })
+            .then(() => {
+                setIsAuth(false);
+                localStorage.clear();
+            })
+        } catch(e) {
+            setAlert({
+                isOpen: true,
+                text: e.message,
+            });
+        };
+    }
+    
+    return (
+        <div className="mainpage-wrapper">
+            <div className="header-wrapper">
+                <h1>
+                    To-Do list
+                </h1>
+                <Button
+                    color="secondary"
+                    size="lg"
+                    onClick={() => logout()}
+                >
+                    Log out
+                </Button>
+            </div>
+            <Button
+                color="primary"
+                size="lg"
+                onClick={() => setIsAddModalOpened(true)}
+            >
+                Add new task
+            </Button>
+            <div className="mainpage-columns-wrapper">
+                <MainPageColumn 
+                    noLeftArrow={true}
+                    name={'Tasks'}
+                    allTasks={allTasks}
+                    setAllTasks={setAllTasks}
+                    tasks={allTasks.filter(elem => elem.stage === 1)}
+                />
+                <MainPageColumn 
+                    name={'In progress'}
+                    allTasks={allTasks}
+                    setAllTasks={setAllTasks}
+                    tasks={allTasks.filter(elem => elem.stage === 2)}
+                />
+                <MainPageColumn 
+                    name={'QA'}
+                    allTasks={allTasks}
+                    setAllTasks={setAllTasks}
+                    tasks={allTasks.filter(elem => elem.stage === 3)}
+                />
+                <MainPageColumn 
+                    noRightArrow={true}
+                    name={'Finished'}
+                    allTasks={allTasks}
+                    setAllTasks={setAllTasks}
+                    tasks={allTasks.filter(elem => elem.stage === 4)}
+                />
+            </div>
+            {isAddModalOpened && <AddTaskModal 
+                isAddModalOpened={isAddModalOpened}
+                setIsAddModalOpened={setIsAddModalOpened}
+                setAllTasks={setAllTasks}
+                allTasks={allTasks}
+            />
+            }
+            <ErrorAlert 
```

Comment:
Not the best way to add Error modal on every page, better to set modal at top level of app and call it via Redux (you can save this, it's recomendation for the future).